access/src/main/external/static/js/admin/src/MoveActionMonitor.js

```diff
@@ -0,0 +1,250 @@
+define('MoveActionMonitor', [ 'jquery'], function($) {
+
+	var defaultOptions = {
+		updateInterval : 5000
+	};
+	
+	function MoveActionMonitor(alertHandler, options) {
+		this.options = $.extend({}, defaultOptions, options);
+		
+		this.alerts = alertHandler;
+		this.moveData = {};
+	}
+	
+	function setDifference(A, B) {
+		var map = {}, C = [];
+
+		for (var i = B.length; i--; )
+			map[B[i]] = null;
+
+		for (var i = A.length; i--; ) {
+			if (!map.hasOwnProperty(A[i]))
+				C.push(A[i]);
+		}
+
+		return C;
+	}
+	
+	MoveActionMonitor.prototype.setResultList = function(resultList) {
+		this.resultList = resultList;
+	};
+	
+	MoveActionMonitor.prototype.activate = function() {
+		this.active = true;
+		
+		// Mark objects that were previously known to be moving
+		var self = this;
+		$.each(this.moveData, function(moveId, moveDetails) {
+			self.markMoving(moveDetails.pids);
+		});
+		
+		// Start refresh loop
+		this.update();
+	};
+	
+	MoveActionMonitor.prototype.update = function() {
+		var self = this;
+		
+		$.ajax({
+			url : "/services/api/listMoves",
+			contentType: "application/json; charset=utf-8",
+			dataType: "json"
+		}).done(function(data) {
+			var remoteMoves = data == null? {} : data;
+			
+			try {
+				// Clean up inactive move operations that have been removed remotely
+				self.cleanMoveTombstones(remoteMoves);
+				// Clean up completed move operations
+				self.completeMoves(remoteMoves);
+			
+				// Indicate new move options
+				self.addMoves(remoteMoves);
+				
+			} finally {
+				// Queue up the next run
+				setTimeout($.proxy(self.update, self), self.options.updateInterval);
+			}
+		}).fail(function() {
+			console.error("Failed to retrieve move information from server");
+			// Keep refreshing alive, but wait a bit longer
+			setTimeout($.proxy(self.update, self), self.options.updateInterval * 2);
+		});
+	};
+	
+	MoveActionMonitor.prototype.cleanMoveTombstones = function(remoteMoves) {
+
+		var localInactive = [];
+		for (var moveId in this.moveData) {
+			if (this.moveData[moveId].inactive) {
+				localInactive.push(moveId);
+			}
+		}
+		
+		if (localInactive.length == 0)
+			return;
+
+		// Cleanable tombstones = Inactive Local Moves - Remote Moves
+		var cleanable = setDifference(localInactive, Object.keys(remoteMoves));
+		
+		if (cleanable.length == 0)
+			return;
+		
+		for (var index in cleanable) {
+			var cleanableId = cleanable[index];
+			delete this.moveData[cleanableId];
+			this.removeUserMove(cleanableId);
+		}
+	};
+	
+	MoveActionMonitor.prototype.completeMoves = function(remoteMoves) {
+
+		// Get the set of move ids which were in progress according to the server
+		var remoteActive = [];
```

Comment:
It is generating a subset of which remove operations are considered to still be in progress on the server side.  The server will respond with moves which have finished, but then hold the moves for another 12 seconds as completed.  So the client determines which moves are newly inactive by subtracting the remotely active moves from the locally active ones.